builtin/arrayview.mbt

```diff
@@ -169,3 +169,144 @@ pub fn rev_foldi[A, B](
     acc
   }
 }
+
+///|
+pub fn map[T, U](self : ArrayView[T], f : (T) -> U) -> ArrayView[U] {
```

Comment:
I am really needing the ability to map an `ArrayView` in my own code, so I wrote this PR: #1321.
After I wrote that, I found this PR, but am not happy about the implementation found here.

I believe that it is _MUCH_ more flexible for the `map*` functions to return an `Array` (not an `ArrayView`).
Then, code that was performing a map on an entire array could easily insert a `[1:]` and now map all but the first entry without modifying any other code.

Additionally, it is much easier to turn an `Array` into an `ArrayView` than to go the other direction.

Reply:
You're pretty right.